### ISSUE TITLE
[ FF7 ] SFX Volume bug

### ISSUE THREAD

**Issue by Bjburn2689** [julianxhokaxhiu/FFNx#744]
This bug is native to the base PC version and not exclusive to FFNx, but I figure it's worth noting any if it can be fixed in the future. There's a steeper drop off in the audibility of music and SFX at lower volume levels in the PC version than on PSX which can render various sound effects inaudible on PC.

**To Reproduce**
Two places I noticed this are

1. After Cloud tells off Barret in the basement of Seventh Heaven, Barret starts rapidly punching the punching bag. His punches are too quiet on PC so you can't hear them.
2. When Tifa falls into the lifestream, a bunch of sounds play to give off the feeling of the countless voices Tifa is hearing in her head. On PC these are inaudible due to the bug so the scene is dead silent until near the end.

Either of these scenes can be looked at to see the issue, or a test case can be made.

**Additional context**
No telling how many scenes this might affect. The above two examples are just ones I noticed.

**Comment by julianxhokaxhiu** [comment]
Thanks for reporting this behavior. Can you please let me know the field names where this happens? So I can look at the field opcodes and try to derive an hypothesis on what is going on.

**Comment by julianxhokaxhiu** [comment]
Root cause of this issue seems to be an hardcoded volume conversion table from the 0-127 scale that you can set in the field opcode to the real -10000 to 0 scale of DirectSound ( used natively in the game code ). The table pretty much uses a scale that apparently doesn't work anymore as expected on generic sound systems and modern Windows ( output may vary, for eg. you can hear something at 5/127 scale when using wired headphones but nothing at all when using bluetooth wired headphones ).

One possible solution here would be to reduce the scale of silence to match the hardware "virtually 0 silence" however DirectSound doesn't offer that.

So in a nutshell, you either use the external layer of FFNx which handles this completely different ( as we use our own custom audio engine ), or we'd have to change the lookup table the game uses to fix generically most of the audio setups ( at the cost of "blasting ears" to the ones using high quality equipment ).

I'm looking at gathering some feedback on which is the "virtual 0 silence" scale that works for most on DirectSound and I'll use that as a starting point for the 0/127 scale.
